### ISSUE TITLE
Feature Request: OpenID connect support

### ISSUE THREAD

**Issue by Lichtjaeger** [simov/grant#85]
OpenID connect is built on top of OAuth2.0.
It has endpoint and claim discovery. With this client, you could reach a much greater variety of providers.
And you could support self-created providers too.

**Comment by simov** [comment]
Thanks for the feedback, @Lichtjaeger, I'll take a look.

**Comment by simov** [comment]
@Lichtjaeger I've published v4.2.0 with some basic support for [OpenID Connect](https://github.com/simov/grant#openid-connect), namely: Grant now tries to parse the returned `id_token` and verifies the `aud` and `nonce` claims. It also supports generating `nonce` for the authorization step.

Full `id_token` validation would require configuration discovery for the provider and its public keys, that needs to be cached and so on. Of course this falls out of scope for Grant. 

However, I made this experimental middleware called [grant-oidc](https://github.com/simov/grant-oidc) that eventually will do just that. For now it fetches the provider configuration and public keys on every authorization attempt.
